supreme/src/commonMain/kotlin/at/asitplus/signum/supreme/validate/CertificateChainValidator.kt

```diff
@@ -0,0 +1,156 @@
+package at.asitplus.signum.supreme.validate
+
+import at.asitplus.signum.CertificateChainValidatorException
+import at.asitplus.signum.CryptoOperationFailed
+import at.asitplus.signum.indispensable.asn1.KnownOIDs
+import at.asitplus.signum.indispensable.asn1.ObjectIdentifier
+import at.asitplus.signum.indispensable.pki.CertificateChain
+import at.asitplus.signum.indispensable.pki.X509Certificate
+import at.asitplus.signum.indispensable.pki.root
+import at.asitplus.signum.supreme.sign.verifierFor
+import at.asitplus.signum.supreme.sign.verify
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+
+val uncheckedCriticalExtensionOids = setOf(
+    KnownOIDs.cRLDistributionPoints_2_5_29_31,
+    KnownOIDs.issuingDistributionPoint_2_5_29_28,
+    KnownOIDs.deltaCRLIndicator,
+)
```

Comment:
Why should we allow certian critical extensions to be unchecked/ignored? Isn't that against the spec?